src/cosalib/aws.py

```diff
@@ -52,14 +52,20 @@ def aws_run_ore_replicate(build, args):
     if len(region_list) == 0:
         raise Exception("no new regions detected")
 
-    source_image = buildmeta['amis'][0]['hvm']
-    source_region = buildmeta['amis'][0]['name']
+    if not args.source_region:
+        args.source_region = buildmeta['amis'][0]['name']
+
+    for a in buildmeta['amis']:
+        if a['name'] == args.source_region:
```

Comment:
Need a check post for-loop that `source_image` isn't empty for if a `source_region` was specified and wasn't found.